lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -2084,13 +2084,6 @@ TypeDecl *EquivalenceClass::lookupNestedType(
           foundMembers);
       for (auto member : foundMembers) {
         if (auto type = dyn_cast<TypeDecl>(member)) {
-          // Resolve the signature of this type.
```

Comment:
@DougGregor there's another place where we do this, and if I remove it, all tests pass, except we no longer diagnose:

```
protocol P {
  typealias T = Self.T
}
```

I guess the next step is to start enabling request evaluator cycle diagnostics...

Reply:
Yeah, I expect we'll learn a lot from experimenting with enabling request-evaluator cycle diagnostics.